governance/xc_admin/packages/xc_admin_common/src/governance_payload/PythGovernanceAction.ts

```diff
@@ -176,10 +179,13 @@ export abstract class PythGovernanceActionImpl implements PythGovernanceAction {
   ): Buffer {
     const headerBuffer = this.header().encode();
 
-    const payloadBuffer = Buffer.alloc(payloadLayout.span);
-    payloadLayout.encode(payload, payloadBuffer);
+    // XXX: there is no way to get the length before the encoding.
```

Comment:
I needed to think a lot why this is happening. I don't like this very much. i see that in `SetDataSources` we haven't defined the whole thing as a layout and manually decode/encode there. Maybe we can do the same?

Reply:
I tried to avoid this but I guess it makes sense to use the approach we already use in `SetDataSources` (I hadn't noticed that before). Updated.